packages/shared/components/popups/ActivityDetailsPopup.svelte

```diff
@@ -1,42 +1,44 @@
 <script lang="typescript">
-    import { Pill, ActivityTypePill, Box, AddressBox, KeyValueBox } from 'shared/components/atoms'
     import { Text, Button } from 'shared/components'
-    import { convertToFiat, currencies, exchangeRates, formatCurrency } from 'shared/lib/currency'
-    import { formatDate, localize } from '@core/i18n'
+    import { localize } from '@core/i18n'
     import { getOfficialExplorer } from 'shared/lib/network'
     import { Platform } from 'shared/lib/platform'
-    import { activeProfile } from 'shared/lib/profile'
-    import { CurrencyTypes } from 'shared/lib/typings/currency'
-    import { Payload } from 'shared/lib/typings/message'
-    import { formatUnitBestMatch } from 'shared/lib/units'
-    import { getMilestoneMessageValue, wallet } from 'shared/lib/wallet'
+    import { getMilestoneMessageValue, isParticipationPayload, wallet } from 'shared/lib/wallet'
     import { FontWeightText } from 'shared/components/Text.svelte'
+    import { TransactionDetails } from 'shared/components/molecules'
     import { getTransactionSubjectAddressOrAccount } from '@lib/utils/transactionObject'
+    import { ActivityType } from '@lib/typings/activity'
 
-    export let id: string
-    export let timestamp: string
-    export let confirmed: boolean
-    export let payload: Payload
-    export let balance: number // migration tx
+    export let message: { id: any; timestamp?: any; confirmed?: any; payload?: any; balance?: any }
+    $: ({ id, payload, balance, timestamp, confirmed } = message)
 
-    let date = localize('error.invalidDate')
+    const { accounts } = $wallet
+    const explorerLink = getOfficialExplorer($accounts[0].clientOptions.network)
+
+    let type: ActivityType
     $: {
-        try {
-            date = formatDate(new Date(timestamp), {
-                dateStyle: 'long',
-                timeStyle: 'medium',
-            })
-        } catch {
-            date = localize('error.invalidDate')
+        if (payload?.type) {
+            if (isParticipationPayload(payload)) {
+                type = ActivityType.StakingTransaction
+            } else if (payload.data.essence.data.internal) {
+                type = confirmed ? ActivityType.Transfer : ActivityType.Transferring
+            } else {
+                type = confirmed
+                    ? payload.data.essence.data.incoming
+                        ? ActivityType.Received
+                        : ActivityType.Sent
+                    : payload.data.essence.data.incoming
+                    ? ActivityType.Receiving
+                    : ActivityType.Sending
```

Comment:
I feel like having an additional else if statement would clear this up.

Reply:
Separated this out in my next PR so that there is only an if statement for the type:
![Screenshot 2022-05-04 at 12 51 24](https://user-images.githubusercontent.com/89913787/166675714-d76a5126-3053-432b-ae3b-cffc2d5dc44f.png)